### PR TITLE
Add slowdown to low temperatures

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Being cold will now slow down player mobs. this increases based on how cold you are, maxing out at the point where you take damage.

## Why / Balance

Cold is a relatively rare threat that has easy counters but doesn't even need to be countered. While heat can quickly damage someone and kill them, cold damage is pretty minor overall and unless you are playing as a lizard, doesn't matter. The less severe stages of cold have even less of an effect.

adding slowdown makes cold a more unique threat while also making space more dangerous while not outright lethal.

## Technical details

because temperature is not predicted, this uses a delayed application. The game replicates the current appropriate slowdown value from the server and then refreshes movement after about a second (enough for the client to hopefully get the new value, god help you if your ping is >1000).

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Being cold now slows down your character.

### CODE REVIEW

File: Content.Shared/Temperature/TemperatureEvents.cs [hunk 1]
```diff
@@ -13,3 +13,18 @@ public ModifyChangedTemperatureEvent(float temperature)
         TemperatureDelta = temperature;
     }
 }
+
+public sealed class OnTemperatureChangeEvent : EntityEventArgs
+{
+    public float CurrentTemperature { get; }
```
Comment (Slava0135) [inline Content.Shared/Temperature/TemperatureEvents.cs:19]: Can't we use `readonly`?

**Comment by deltanedas** [comment]
become faster when on fire when

**Comment by Cojoke-dot** [comment]
Mmm... The effect for the smaller amount of fresorium(if that's how you spell it) will now stop you I think? IDK but you may want to test that.

**Comment by Partmedia** [comment]
It looks like this uses TemperatureComponent's `CurrentTemperature`. Keep in mind that when applied to mobs, there is also `ThermalRegulatorComponent` and thermal resistance to the outside due to clothing modifiers so this slowdown will probably significantly lag actually going outside.

**Comment by EmoGarbage404** [comment]
> It looks like this uses TemperatureComponent's `CurrentTemperature`. Keep in mind that when applied to mobs, there is also `ThermalRegulatorComponent` and thermal resistance to the outside due to clothing modifiers so this slowdown will probably significantly lag actually going outside.

Its my understanding that all of those systems are handled by modifying changes to CurrentTemperature.

**Comment by Plykiya** [comment]
> become faster when on fire when

is this the buff that elite suit nukies need?

I wonder what kind of implications the slowdown has on emergency situations where people are in emergency EVA suits walking on the outside of the station

**Comment by deltanedas** [comment]
i was imagining more whos your daddy baby on fire going mach 3

File: Content.Shared/Temperature/Components/TemperatureSpeedComponent.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public float? CurrentSpeedModifier;
+
+    /// <summary>
+    /// The time at which the temperature slowdown is updated.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public TimeSpan? NextSlowdownUpdate;
+}
```
Comment (metalgearsloth) [inline Content.Shared/Temperature/Components/TemperatureSpeedComponent.cs:30]: autopausedfield?

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**